syntax/printer.go

```diff
@@ -269,16 +269,17 @@ func (p *Printer) space() {
 }
 
 func (p *Printer) spacePad(pos Pos) {
-	if p.cols.lineStart {
-		// Never add padding at the start of a line, since this may
+	if p.cols.lineStart && (!p.keepPadding || p.indentSpaces == 0) {
+		// Never add padding at the start of a line unless keepPadding
+		// is set and we are indenting with spaces, since this may
 		// result in broken indentation or mixing of spaces and tabs.
```

Comment:
The only result we're avoiding here is mixing spaces and tabs now, so you should remove the bit about "broken indentation". When I wrote that, I did indeed mean obeying the user's choice of space indentation, as opposed to enforcing the equivalent of tabs.